python/metatensor-torch/metatensor/torch/__init__.py

```diff
@@ -30,20 +31,39 @@
 if HAS_METATENSOR_OPERATIONS:
     from . import operations  # noqa
     from .operations import *  # noqa
+else:
+    MISSING_SUBPACKAGES.append("metatensor-operations")
+
+
+try:
+    import metatensor.learn  # noqa
+
+    HAS_METATENSOR_LEARN = True
+except ImportError:
+    HAS_METATENSOR_LEARN = False
 
 
+if HAS_METATENSOR_LEARN:
+    from . import learn  # noqa
+    from .learn import *  # noqa
 else:
+    MISSING_SUBPACKAGES.append("metatensor-learn")
+
+
+if not (HAS_METATENSOR_LEARN) or not (HAS_METATENSOR_OPERATIONS):
     # __getattr__ is called when a symbol can not be found, we use it to give
-    # the user a better error message if they don't have metatensor-operations
+    # the user a better error message if they don't have all metatensor subpackages
     def __getattr__(name):
         raise AttributeError(
-            f"metatensor.torch.{name} is not defined, are you sure you have the "
-            "metatensor-operations package installed?"
+            f"metatensor.torch.{name} is not defined, you might have not "
+            "installed the corresponding metatensor subpackage "
+            f"({' or '.join(MISSING_SUBPACKAGES)})."
         )
 
 
 __all__ = [
     "Labels",
+    "LabelsEntry",
```

Comment:
`LabelsEntry` not being in `__all__` was intentional, why do you need it here?

Reply:
We need it later in TensorModules

Reply:
Even if you need it in ModuleMap, it does not need to be in `__all__`. `__all__` controls what's imported when a user does `from metatensor.torch import *`, and I don't see a point for `LabelsEntry` here.

Reply:
Even if it's not in `__all__`, you can still do `from metatensor.torch import LabelsEntry`

Reply:
removed from __all__